src/auth/user/user.service.ts

```diff
@@ -46,7 +54,8 @@ export class UserService {
       Logger.error('Error getting user', error);
     }
   }
-  async verifyUser(email: string, password: string) {
+
+  async verifyAndGetUser(email: string, password: string) {
```

Comment:
It should be verifyUser. not verifyAndGetUser, it's confusing. 
Also it should fit, Single Responsibility Principle.